db/migrate/20190125170630_add_reset_password_token_index_to_spree_users.rb

```diff
@@ -0,0 +1,8 @@
+class AddResetPasswordTokenIndexToSpreeUsers < SolidusSupport::Migration[4.2]
+  def change
+    Spree::User.reset_column_information
+    if Spree::User.column_names.include?('reset_password_token') && !index_exists?(:spree_users, :reset_password_token)
```

Comment:
I don't think this migration can be rolled back since `!index_exists?(:spree_users, :reset_password_token)` will always return true rolling back, not allowing to execute the schema change, right?

Reply:
@kennyadsl you're right. 

The problem is we don't know if somebody already added the index, so we need to check. 
Also, if somebody already added the index, we should not remove it when rolling back. I can't think of a solution that would ensure that the index, in case of rollback, is removed only if it was not present before.

But maybe  the next best thing is to remove the index on rollbacks, even when it was already present. After all, those who already have it should be a minority. I think I can update the PR and fix it by setting explicit `up` and `down`.

Reply:
What about creating an index with a different name then the default one if it does not exist? 

This way we can rollback safely since when that index is present we are sure it has been added with the second migration.

Reply:
@kennyadsl that would work, but having non-standard index name may be confusing/surprising for everyone else. 

Is it worth? The annoyance for the large majority may outweigh the benefits for a very small minority. And this *may* be an issue only when rolling back.
So I'm not sure about this... what do you think?

Reply:
I don't know - I'm not sure how an index name could be confusing. I don't frequently end up looking at index names, but it could be me. We could document why that "non-standard" index name is there via a code comment maybe? 

On the other side, this issue would happen only when you roll back **both** migrations which is kinda unrealistic. 

I don't have strong opinions here. I trust your judgment on what's the best solution.

Reply:
@kennyadsl I eventually decided in favor of the custom index name. I agree the index name annoyance is very minimal, and I prefer to be as nice as possible to all installations.

The custom name is a bit long at 61 chars but it should be ok for all major RDBMS.